ddtrace/span.py

```diff
@@ -101,13 +97,40 @@ def __init__(
         self.parent_id = parent_id
         self.tracer = tracer
 
-        # sampling
-        self.sampled = True
-
-        self._context = context
         self._parent = None
         self._ignored_exceptions = None  # type: Optional[List[Exception]]
 
+    @property
+    def context(self):
+        # type: () -> Context
+        from .tracer import _get_trace
+
+        trace = _get_trace(self.trace_id)
+        return Context(
+            trace_id=self.trace_id,
+            span_id=self.span_id,
```

Comment:
is this correct? don't we need to know the current span, not this spans span id?

Reply:
I'm picturing that `Context` is now basically a snapshot of an active trace (same information required to do distributed tracing).

You're right that if we were to pass around `Context` objects within the same process that we'd probably want to have the span object itself so that we can set `span._parent` (which we need to inherit the service name) but I don't think there's a case where we'd use `Context` locally.

I'd also argue we want to move away from even having `span._parent` in the long term.